---
base_link: world
# =====================================================
# Note: these parameters are copied from Robotiq gripper without tweaking

object:
  ideal_grasp_pose: [0, 0, 0, -1.57, 0, 0]  # Pose of the TCP to pick the object
  end_effector_name: end_effector
  object_description: $(find moveit_grasp_b_tools)/urdf/test.urdf
  initial_pose: [0.6, 0.7, 1.1, 0, 0, 0]
  grasp_point: point_a
  subframes:
    - name: point_a
      pose: [0, 0, 0.025, 0, 0, 0]

  approach:
    direction: [0.0, 0.0, -1.0]
    desired_distance: 0.1
    minimum_distance: 0.05

  retreat:
    direction: [0.0, 0.0, 1.0]
    desired_distance: 0.2
    minimum_distance: 0.05

  gripper_closure:
    - joint_name: gripper_finger1_joint
      position: 0.5

  gripper_oppening:
    - joint_name: gripper_finger1_joint
      position: 0.0

  tcp_to_eef_mount_transform: [0, 0, -0.15, 0, 0, 0]

end_effector:
  end_effector_name: end_effector  # ee group name
  end_effector_type: suction  # 'finger' or 'suction'

  grasp_depth_resolution: 0.04

  # A grasp motion is composed of three linear paths: approach, retreat,
  # lift The values below define the desired distances of each path.  Length
  # of approach path (this is in addition to the grasp_max_depth)
  # approach path (in addition to grasp_max_depth)
  approach_distance_desired: 0.05
  # retreat path (in addition to grasp_max_depth)
  retreat_distance_desired: 0.05
  # lift path (used as MINIMUM safety distance from surface)
  lift_distance_desired: 0.02

  # minimum padding on each side of the object on approach
  grasp_padding_on_approach: 0.005  # meter

  # Joint names and state values for the end effector.
  # The names should match the values of pregrasp_posture and grasp_posture.
  # These values can be left empty if end_effector_type is 'suction'
  pregrasp_posture: []  #  open position (not used in suction)
  grasp_posture: []  #  close position (not used in suction)
  joints: []  #(not used in suction)

  ##############################
  ##  Finger gripper parameters
  ##############################

  # Distances between fingers matching to pregrasp_posture and
  # grasp_posture, in meters
  max_finger_width: 0.085
  min_finger_width: 0.06

  # Maximum allowed finger width for a grasp.
  # This value should be considerably smaller than max_finger_width
  # to allow padded collision checks
  max_grasp_width: 0.08

  # width of gripper fingers to ensure the finger tips fully overlap the
  # object
  gripper_finger_width: 0.015

  ##############################
  ## Suction gripper parameters
  ##############################
  # The width of the suction gripper along the specified axes. (Assumes that
  # Z points in the direction of the suction cups)
  active_suction_range_x: 0.01
  active_suction_range_y: 0.01

  # Some suction grippers can selectively enable and disable regions of
  # suction. If you have more than one region that can be selectively
  # activated or de-activated, set those values here and moveit_grasps will
  # prioritize grasp candidates which use full suction regions rather than
  # partial regions.  it will also set the suction_regions_in_contact_
  # variable in the grasp candidates so that you are able to only enable
  # suction on regions that are in contact with the grasp target.  Note that
  # this metric assumes that suction regions are equally sized ie. width =
  # active_suction_range_x / suction_cols_count
  #
  suction_rows_count: 1
  suction_cols_count: 1
